### PR TITLE
Website: add simple status badge component

### PR DESCRIPTION
**Pull request by Schmavery**
Adds component that queries status.codaprotocol.com for the status of specific services.

This probably warrants some kind of a design pass but should contain all the wiring necessary to build a component like this.

### CODE REVIEW

File: frontend/website/pages/docs/my-first-transaction.mdx [hunk 1]
```diff
@@ -11,6 +12,8 @@ In this section, we'll make our first transaction on the Coda network. After [in
 
   You can find the current status of the network and related services here: [status.codaprotocol.com](https://status.codaprotocol.com)
 
+  Testnet network status: <StatusBadge service="Network"/>
```
Comment (o1pranay) [inline frontend/website/pages/docs/my-first-transaction.mdx]:
```suggestion
  Testnet status: <StatusBadge service="Network"/>
```

feels redundant to say net network?

File: frontend/website/src/components/StatusBadge.re [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+  name: string,
+  status: string,
+};
+type response = {components: array(component)};
+
+external parseStatusResponse: Js.Json.t => response = "%identity";
+
+type service = [ | `Summary | `Network | `Faucet | `EchoBot | `GraphQLProxy];
+
+type status =
```
Comment (o1pranay) [inline frontend/website/src/components/StatusBadge.re]: these are the other states fyi
Screen Shot 2020-02-12 at 7 49 56 PM

File: frontend/website/src/components/StatusBadge.re [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    Unknown;
+  };
+
+let parseServiceName = name =>
+  switch (name) {
+  | "Network" => `Network
+  | "Faucet" => `Faucet
+  | "Echo Bot" => `EchoBot
+  | "GraphQL Proxy" => `GraphQLProxy
+  | "Coda Testnet"
```
Comment (o1pranay) [inline frontend/website/src/components/StatusBadge.re]: think we can remove this one?

File: frontend/website/src/components/StatusBadge.re [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+      |> Promise.map(parseStatusResponse)
+      |> Promise.iter(response => {
+           let components =
+             response.components
+             |> Array.to_list
+             |> List.filter(c => parseServiceName(c.name) == service);
+           switch (components) {
+           | [] => Js.Console.warn("Error retrieving status")
+           | [{status}, ..._] =>
+             Js.log2("test", status);
```
Comment (o1pranay) [inline frontend/website/src/components/StatusBadge.re]: nit remove

File: frontend/website/src/components/StatusBadge.re [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+             setStatus(_ => parseStatus(status));
+           };
+         });
+      None;
+    });
+    let (statusStr, bgColor) =
+      switch (status) {
+      | Unknown => ("Unknown", Theme.Colors.grey)
+      | Operational => ("Operational", Theme.Colors.clover)
+      | Maintenance => ("Under Maintenance", Theme.Colors.rosebud)
```
Comment (o1pranay) [inline frontend/website/src/components/StatusBadge.re]: nit change color to shade of blue to be more similar to the status page color codes

**Comment by figitaki** [comment]
Closes #4397

**Pull request merged**